src/NHibernate/Proxy/AbstractLazyInitializer.cs

```diff
@@ -287,9 +286,12 @@ private void SetReadOnly(bool readOnly)
 			if (initialized)
 			{
 				EntityKey key = GenerateEntityKeyOrNull(_id, _session, _entityName);
-				if (key != null && _session.PersistenceContext.ContainsEntity(key))
+				if (key != null)
 				{
-					_session.PersistenceContext.SetReadOnly(_target, readOnly);
+					if (persistenceContext.ContainsEntity(key))
```

Comment:
Seems a bit undue `if` split in two `if`. (Maybe a first change was to introduce an intermediate variable, but which was later removed due to the method signature change, without recombining the `if`.)